core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -152,13 +156,19 @@ class KafkaApis(val requestChannel: RequestChannel,
     // ensureTopicExists is only for client facing requests
     // We can't have the ensureTopicExists check here since the controller sends it as an advisory to all brokers so they
     // stop serving data to clients for the topic being deleted
-    val stopReplicaRequest = request.requestObj.asInstanceOf[StopReplicaRequest]
+    val stopReplicaRequest = request.body.asInstanceOf[StopReplicaRequest]
 
-    authorizeClusterAction(request)
```

Comment:
@ijuma Indeed, I have followed what has been done @granthenke. I would also add that using `authorizeClusterAction` doesn't work anyway because it throws `k.c.ClusterAuthorizationException` which will be considered as UNKNOWN by `request.body.getErrorResponse(request.header.apiVersion, e)` in KafkaApi.scala:98.

Another way would be to use `Errors.CLUSTER_AUTHORIZATION_FAILED.maybeThrow` if you prefer.

WDYT?